### PR TITLE
Modernize icons

### PR DESCRIPTION
**Pull request by DuMaM**
This PR add new svg icon from set used across whole Jenkins -> https://ionic.io/ionicons
This follow with more modernize Jenkins look and enable theming.
More here: https://weekly.ci.jenkins.io/design-library/Symbols/

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Other connected PR's:
https://github.com/jenkinsci/workflow-cps-plugin/pull/559
Note: This require latest LTS as a base: 2.346.1

Project Status:
FireShot Capture 009 - qmk  Jenkins  - 192 168 0 31

Build Status:
FireShot Capture 010 - qmk #178  Jenkins  - 192 168 0 31

Project Side Panel:
FireShot Capture 011 - autoupdate  Jenkins  - 192 168 0 31

Build Side Panel:
FireShot Capture 012 - autoupdate #193  Jenkins  - 192 168 0 31

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly [hunk 1]
```diff
@@ -28,24 +28,24 @@
...
                 <j:otherwise>
-                    <l:task icon="icon-terminal icon-md" href="${buildUrl.baseUrl}/console" title="${%Console Output}">
-                        <l:task href="${buildUrl.baseUrl}/consoleText" icon="icon-document icon-md" title="${%View as plain text}"/>
+                    <l:task icon="symbol-terminal icon-md" href="${buildUrl.baseUrl}/console" title="${%Console Output}">
+                        <l:task href="${buildUrl.baseUrl}/consoleText" icon="symbol-document-text icon-md" title="${%View as plain text}"/>
                     </l:task>
                 </j:otherwise>
             </j:choose>
-            <l:task icon="icon-notepad icon-md" href="${buildUrl.baseUrl}/configure" title="${h.hasPermission(it,it.UPDATE)?'%Edit Build Information':'%View Build Information'}"/>
+            <l:task icon="symbol-details icon-md" href="${buildUrl.baseUrl}/configure" title="${h.hasPermission(it,it.UPDATE)?'%Edit Build Information':'%View Build Information'}"/>
```
Comment (DuMaM) [inline src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly:45]: I'm only wondering if we should use an edit symbol or add custom one like this one
https://ionic.io/ionicons
create-outline

**Review by jglick — CHANGES_REQUESTED** [review]
> This require latest LTS as a base: 2.346.1

Then update `jenkins.version` and the `bom` accordingly: https://github.com/jenkinsci/workflow-cps-plugin/pull/559#issuecomment-1176536065

File: src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly (hunk 1 above)
Comment (DuMaM) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly]: @jglick what do you think about this icon?

**Comment by jglick** [comment]
Maybe @timja or someone more familiar with UI work wants to review these PRs?

**Comment by NotMyFault** [comment]
I'll take a look later.

**Review by NotMyFault — APPROVED** [review]
Looks nice, thanks

File: src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly (hunk 1 above)
Comment (DuMaM) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly]: @NotMyFault What do you think about using any of those symbols in "Edit Build Description" task:
-> edit symbol https://github.com/jenkinsci/jenkins/blob/master/war/src/main/resources/images/symbols/edit.svg
-> details symbol https://github.com/jenkinsci/jenkins/blob/master/war/src/main/resources/images/symbols/details.svg
-> custom one placed above

For me custom one fits the best here, but i do not want to introduce new symbols in one of the most used plugin, when there is no such need.

Comment (NotMyFault) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly]: That should probably be a gear symbol to align with the design used in other job types.

The current "Edit Build Information" icon should be the Status icon as well, that's how it's done in other job types.

Comment (DuMaM) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly]: I don't agree about gear symbol. It's overused in Jenkins (also by other plugins). 
Previously it was a notepad icon, so this is another argument against it.

> The current "Edit Build Information" symbol should be made the Status symbol as well, that's how it's done in other job types.
Could you provide more details about it?

Comment (DuMaM) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly]: Ok, so details symbol is big no no here (Edit Build Description), and i should also update status icon from:
symbol-search => symbol-details

Do I understand correctly?

Comment (NotMyFault) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly]: Yes, take a look at a freestyle job for the design in place for a better comparison to a pipeline job.

Comment (DuMaM) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly]: Ok I applied those changes

File: src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowJob/sidepanel.jelly [hunk 2]
```diff
@@ -31,14 +31,14 @@
             <j:set var="url" value="${h.getNearestAncestorUrl(request,it)}"/>
             <j:choose>
                 <j:when test="${it.parent==app}">
-                    <l:task icon="icon-up icon-md" href="${rootURL}/" title="${%Back to Dashboard}" contextMenu="false"/>
+                    <l:task icon="symbol-arrow-up" href="${rootURL}/" title="${%Back to Dashboard}" contextMenu="false"/>
```
Comment (uhafner) [inline src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowJob/sidepanel.jelly:34]: Why is this link present anyway? It links to the a page that can be accessed with the breadcrumbs.  It introduces clutter in the side panel. (The same is wrong in Jenkins core.) See discussion in https://github.com/jenkinsci/jenkins/pull/6783

File: src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowJob/sidepanel.jelly [hunk 3]
```diff
@@ -31,14 +31,14 @@
...
                 </j:when>
                 <j:otherwise>
-                    <l:task icon="icon-up icon-md" href="${url}/../../" title="${%Up}" contextMenu="false"/>
+                    <l:task icon="symbol-arrow-up" href="${url}/../../" title="${%Up}" contextMenu="false"/>
                 </j:otherwise>
             </j:choose>
-            <l:task icon="icon-search icon-md"  href="${url}/" title="${%Status}" contextMenu="false"/>
-            <l:task icon="icon-notepad icon-md" href="${url}/changes" title="${%Changes}"/>
+            <l:task icon="symbol-notepad"  href="${url}/" title="${%Status}" contextMenu="false"/>
```
[the remark this answers is not in the document]
Comment (NotMyFault) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowJob/sidepanel.jelly:40]: There's no other sidepanel link that takes you to the job overview if you want to head back from a different view, e.g. viewing tests, no?

**Comment by uhafner** [comment]
The Pipeline Steps link uses the wrong icon, the gear should be reserved to the configuration.

**Comment by uhafner** [comment]
The new icon should be part of core. It does not make sense to define new icons here that should be reused by other similar jobs.

File: src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowJob/sidepanel.jelly (hunk 3 above)
[the remark this answers is not in the document]
Comment (uhafner) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowJob/sidepanel.jelly:40]: No, we do not have those links. The breadcrumbs should be used to navigate the hierarchy.

File: src/main/resources/images/symbols/edit-notepad.svg [hunk 4]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" class="ionicon" viewBox="0 0 512 512"><title>Create</title><path d="M384 224v184a40 40 0 01-40 40H104a40 40 0 01-40-40V168a40 40 0 0140-40h167.48" fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="32"/><path d="M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z"/></svg>
```
Comment (DuMaM) [inline src/main/resources/images/symbols/edit-notepad.svg:1]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" class="ionicon" viewBox="0 0 512 512">
  <title>edit-notepad</title>
  <path d="M384 224v184a40 40 0 01-40 40H104a40 40 0 01-40-40V168a40 40 0 0140-40h167.48" fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="32"/>
  <path d="M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z" fill="currentColor"/>
</svg>
```

**Comment by DuMaM** [comment]
> `symbol-notepad` doesn't exist, should be details, that's the correct override.
> 
> Screenshot 2022-07-07 at 12 57 06
> 
> The pencil of the added symbol isn't theme able. Make sure its stroke and filling inherit `currentColor`, that's what the symbol API uses to apply a design.

Done, I also changed title to be more acurate.

**Comment by DuMaM** [comment]
> The Pipeline Steps link uses the wrong icon, the gear should be reserved to the configuration.

That's true, but it looks like this is located in other plugin.
https://github.com/jenkinsci/jenkins/blob/master/war/src/main/resources/images/symbols/build-steps.svg
This should be fine.

File: src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly [hunk 5]
```diff
@@ -28,24 +28,24 @@
...
                 <j:otherwise>
-                    <l:task icon="icon-terminal icon-md" href="${buildUrl.baseUrl}/console" title="${%Console Output}">
-                        <l:task href="${buildUrl.baseUrl}/consoleText" icon="icon-document icon-md" title="${%View as plain text}"/>
+                    <l:task icon="symbol-terminal" href="${buildUrl.baseUrl}/console" title="${%Console Output}">
+                        <l:task href="${buildUrl.baseUrl}/consoleText" icon="symbol-document-text" title="${%View as plain text}"/>
                     </l:task>
                 </j:otherwise>
             </j:choose>
-            <l:task icon="icon-notepad icon-md" href="${buildUrl.baseUrl}/configure" title="${h.hasPermission(it,it.UPDATE)?'%Edit Build Information':'%View Build Information'}"/>
+            <l:task icon="symbol-edit-notepad plugin-workflow-job" href="${buildUrl.baseUrl}/configure" title="${h.hasPermission(it,it.UPDATE)?'%Edit Build Information':'%View Build Information'}"/>
```
[the remark this answers is not in the document]
Comment (DuMaM) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly:45]: Sorry I'm not too familiar with permissions setup in jelly. Do you have any examples?
I guess that you want me to hide this task from users without update permissions, but why do we should change this icon in such case?

[the remark this answers is not in the document]
Comment (DuMaM) [reply src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly]: Done :)

**Comment by DuMaM** [comment]
> Please do not forget to create a followup PR In core so that freestyle jobs will use the same icons for the same operations.

Ok. I will. Now it's using a gear icon what is not perfect.

**Review by NotMyFault — APPROVED** [review]

File: src/main/java/org/jenkinsci/plugins/workflow/job/views/FlowGraphTableAction.java [hunk 6]
```diff
@@ -42,7 +42,7 @@ private FlowGraphTableAction(WorkflowRun run) {
     }
 
     @Override public String getIconFileName() {
-        return "gear.png";
+        return "symbol-build-steps";
```
[the remark this answers is not in the document]
Comment (uhafner) [reply src/main/java/org/jenkinsci/plugins/workflow/job/views/FlowGraphTableAction.java:45]: We should embed the icon from https://github.com/jenkinsci/jenkins/pull/6485 here (in the same way as edit-note.svg) and remove it if a new LTS baseline has been selected.

[the remark this answers is not in the document]
Comment (NotMyFault) [reply src/main/java/org/jenkinsci/plugins/workflow/job/views/FlowGraphTableAction.java:45]: > 

The "x" implies jenkins can't resolve it, source: https://github.com/jenkinsci/jenkins/blob/stable-2.346/core/src/main/java/org/jenkins/ui/icon/IconSet.java#L65

That would be the case if a plugin uses a symbol- core is not awaer of.

[the remark this answers is not in the document]
Comment (DuMaM) [reply src/main/java/org/jenkinsci/plugins/workflow/job/views/FlowGraphTableAction.java:45]: Oh... I didn't check this. It looks like it was added today...
https://github.com/jenkinsci/jenkins/commit/fe410274a0de6ed13ed2674a92bc6641c4adf239
😵‍💫

**Review by jglick — APPROVED** [review]

**Pull request merged**